packages/constants/noobaa.ts

```diff
@@ -22,31 +27,39 @@ export const NS_PROVIDERS_NOOBAA_MAP = _.pick(
   BC_PROVIDERS.S3,
   BC_PROVIDERS.AZURE,
   BC_PROVIDERS.IBM,
-  BC_PROVIDERS.FILESYSTEM,
+  BC_PROVIDERS.FILESYSTEM
 );
 
 export const NOOBAA_TYPE_MAP = {
-  [BC_PROVIDERS.AWS]: 'aws-s3' as const,
-  [BC_PROVIDERS.S3]: 's3-compatible' as const,
-  [BC_PROVIDERS.AZURE]: 'azure-blob' as const,
-  [BC_PROVIDERS.GCP]: 'google-cloud-storage' as const,
-  [BC_PROVIDERS.PVC]: 'pv-pool' as const,
-  [BC_PROVIDERS.IBM]: 'ibm-cos' as const,
-  [BC_PROVIDERS.FILESYSTEM]: 'nsfs' as const,
+  [BC_PROVIDERS.AWS]: "aws-s3" as const,
+  [BC_PROVIDERS.S3]: "s3-compatible" as const,
+  [BC_PROVIDERS.AZURE]: "azure-blob" as const,
+  [BC_PROVIDERS.GCP]: "google-cloud-storage" as const,
+  [BC_PROVIDERS.PVC]: "pv-pool" as const,
+  [BC_PROVIDERS.IBM]: "ibm-cos" as const,
+  [BC_PROVIDERS.FILESYSTEM]: "nsfs" as const
 };
 
 export const NS_NOOBAA_TYPE_MAP = _.pick(
   NOOBAA_TYPE_MAP,
   BC_PROVIDERS.AWS,
   BC_PROVIDERS.S3,
   BC_PROVIDERS.AZURE,
-  BC_PROVIDERS.IBM,
+  BC_PROVIDERS.IBM
 );
 
+export const NS_PROVIDER_REVERSE_MAP = {
+  [NOOBAA_TYPE_MAP[BC_PROVIDERS.AWS]]: BC_PROVIDERS.AWS as const,
+  [NOOBAA_TYPE_MAP[BC_PROVIDERS.AZURE]]: BC_PROVIDERS.AZURE  as const,
+  [NOOBAA_TYPE_MAP[BC_PROVIDERS.S3]]: BC_PROVIDERS.S3 as const,
+  [NOOBAA_TYPE_MAP[BC_PROVIDERS.IBM]]: BC_PROVIDERS.IBM as const,
+  [NOOBAA_TYPE_MAP[BC_PROVIDERS.FILESYSTEM]]: BC_PROVIDERS.FILESYSTEM as const  
+}
```

Comment:
NSS does not support GCP ??

Reply:
add everthing just like `NOOBAA_TYPE_MAP` and use only the once which is required.